amr-wind/wind_energy/ABLFieldInit.cpp

```diff
@@ -341,18 +418,44 @@ void ABLFieldInit::init_tke(
         const auto& bx = mfi.tilebox();
         const auto& tke = tke_fab.array(mfi);
 
-        // Profile definition from Beare et al. (2006)
-        amrex::ParallelFor(
-            bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
-                const amrex::Real z = problo[2] + (k + 0.5) * dx[2];
+        if (m_initial_wind_profile) {
+            const int ntvals = static_cast<int>(m_wind_heights.size());
+            const amrex::Real* windh = m_windht_d.data();
+            const amrex::Real* tke_data = m_prof_tke_d.data();
+            amrex::ParallelFor(
+                bx, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                    const amrex::Real z = problo[2] + (k + 0.5) * dx[2];
+
+                    amrex::Real tke_prof = tke_data[0];
+
+                    for (int iz = 0; iz < ntvals - 1; ++iz) {
+                        if ((z > windh[iz]) && (z <= windh[iz + 1])) {
```

Comment:
bisection search here too? looks like this is also just linear interpolation

Reply:
Called once only when we start precursor simulations. Did not want to make any changes here.